file/cache.lisp

```diff
@@ -15,9 +19,16 @@
 (defclass file-cache-stream (file-stream)
   ((%direction :initarg :direction :reader direction)
    (%buffer :initarg :buffer :accessor buffer)
+   (%dirty-block :initarg :dirty-block :accessor dirty-block)
+   (%dirty-block-n :initarg :dirty-block-n :accessor dirty-block-n)
    (%position :initarg :position :accessor file-position*)
-   (%length :initarg :length :accessor file-length*))
-  (:default-initargs :position 0 :length 0))
+   (%length :initarg :length :accessor file-length*)
+   (%block-length :initarg :block-length :accessor block-length))
+  (:default-initargs :buffer (make-hash-table :weakness :key-and-value)
```

Comment:
Same issue here. Weak key-and-value hash-table with a fixnum key is equivalent to a weak value hash-table.